frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -65,9 +66,61 @@ const ct = composebox_typeahead;
 ct.__Rewire__("max_num_items", 15);
 
 const mention_all = ct.broadcast_mentions()[0];
+const mention_everyone = ct.broadcast_mentions()[1];
+const mention_stream = ct.broadcast_mentions()[2];
 assert.equal(mention_all.email, "all");
 assert.equal(mention_all.full_name, "all");
+assert.equal(mention_everyone.email, "everyone");
+assert.equal(mention_everyone.full_name, "everyone");
+assert.equal(mention_stream.email, "stream");
+assert.equal(mention_stream.full_name, "stream");
+
+run_test("verify wildcard mentions typeahead for stream message", () => {
+    const wildcard_string = $t({defaultMessage: "Notify stream"});
+
+    assert.equal(
+        mention_all.special_item_text,
+        $t(
+            {defaultMessage: "{wildcard_mention_token} ({wildcard_string})"},
+            {wildcard_mention_token: mention_all.email, wildcard_string},
+        ),
+    );
+    assert.equal(
+        mention_everyone.special_item_text,
+        $t(
+            {defaultMessage: "{wildcard_mention_token} ({wildcard_string})"},
+            {wildcard_mention_token: mention_everyone.email, wildcard_string},
+        ),
+    );
+    assert.equal(
+        mention_stream.special_item_text,
+        $t(
+            {defaultMessage: "{wildcard_mention_token} ({wildcard_string})"},
+            {wildcard_mention_token: mention_stream.email, wildcard_string},
+        ),
+    );
+});
 
+run_test("verify wildcard mentions typeahead for private message", () => {
+    const wildcard_string = $t({defaultMessage: "Notify recipients"});
+    if (compose_state.get_message_type() === "private") {
```

Comment:
This if statement causes the code below to not run. What you want to do is the test setup work required to make that the current state, which I'll do in my revision.